index.d.ts

```diff
@@ -1,10 +1,16 @@
 declare module "@iarna/toml" {
+  type AnyJson = boolean | number | string | null | JsonArray | JsonMap
```

Comment:
Does `number` match `bigint`? (`typeof 123n` === 'bigint')

Missing from here is: datetime (and I don't know how the distinct, date, time and local datetime types would be represented).

Reply:
Typescript, standing as a super-set of javascript, has no primitive type for `bigint`, so `number` is just fine.

We **cannot** map all TOML types to typescript types due to the language limitation. However, there is an easy way to know how we can define a type, that is to inspect all possible values in javascript. Since typescript has **the same primitive types** as javascript, we can describe the types according to the javascript values.

Type for `datetime` should be a good example. Since in javascript, a TOML datetime is interpreted into a Date object, then we can define that type `AnyJson`  should also be a `Date`. 
(i.e. `type AnyJson = boolean | number | string | null | JsonArray | JsonMap | `**`Date`**)

```js
const toml = require('@iarna/toml')
const {  time } = toml .parse('time = 1979-05-27T00:32:00.999999-07:00')

time // 1979-05-27T07:32:00.999Z
typeof time // "object"
time instanceof Date // true
```

Reply:
`bigint` is JavaScript and a primitive type. `number` is not `bigint`. 

https://developers.google.com/web/updates/2018/05/bigint#a_new_primitive

`bigint`s are supported by this library if your JS implementation has gotten around to implementing them. Maybe you can't use them with TypeScript yet though? ¯\\\_(ツ)\_/¯

Reply:
`bigint` will land in Typescript by November, 2018.

Roadmap: https://github.com/Microsoft/TypeScript/wiki/Roadmap#32-november-2018
Watch this thread for update: https://github.com/Microsoft/TypeScript/issues/15096

I'll use a workaround for this currently. That is, to alias `BigInt` to `number` as mentioned [here](https://github.com/Microsoft/TypeScript/issues/15096#issuecomment-400186862).
But it just not a perfect workaround since js or ts has no operator overloading, this should by solved by November. 😄